abce/trade.pyx

```diff
@@ -382,7 +382,7 @@ class Trade:
             quantity = available
 
         offer_id = self._offer_counter()
-        self._haves[good] -= quantity
+        self._haves.destroy(good, quantity)
```

Comment:
We are calling python from cython, that might makes stuff slower, we need to check this though. Maybe inventory should be in cython? Probably not as the dictionary call Cann not be in cython anyway.